src/Storage/Factory/EntityFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace EMS\CommonBundle\Storage\Factory;
+
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use EMS\CommonBundle\Storage\Service\EntityStorage;
+use EMS\CommonBundle\Storage\Service\StorageInterface;
+use Psr\Log\LoggerInterface;
+
+class EntityFactory implements StorageFactoryInterface
+{
+
+    /** @var LoggerInterface */
+    private $logger;
+    /** @var bool */
+    private $registered = false;
+    /** @var Registry */
+    private $doctrine;
+
+    public function __construct(LoggerInterface $logger, Registry $doctrine)
+    {
+        $this->logger = $logger;
+        $this->doctrine = $doctrine;
+    }
+
+    public function createService(array $parameters): ?StorageInterface
+    {
+        if ('db' !== $parameters['type'] ?? null) {
```

Comment:
Make constants of the Types in each file? 

private CONST STORAGE_TYPE='db';

That way every file has the same way of defining what to put in the configuration. Maybe make this public to exploit it in php config files (if applicable)

Reply:
Done